flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientITCase.java

```diff
@@ -153,89 +130,98 @@ static void setup(@InjectClusterClientConfiguration Configuration configuration)
-        String in = getInputFromPath(sqlPath);
-        List<TestSqlStatement> testSqlStatements = parseSqlScript(in);
-        List<String> sqlStatements =
-                testSqlStatements.stream().map(s -> s.sql).collect(Collectors.toList());
-        List<Result> actualResults = runSqlStatements(sqlStatements, configuration);
-        String out = transformOutput(testSqlStatements, actualResults);
-        assertThat(out).isEqualTo(in);
-    }
+                "$VAR_JOBMANAGER_RPC_ADDRESS", MINI_CLUSTER.getClientConfiguration().get(ADDRESS));
 
-    /**
-     * Returns printed results for each ran SQL statements.
-     *
-     * @param statements the SQL statements to run
-     * @return the printed results on SQL Client
-     */
-    private List<Result> runSqlStatements(List<String> statements, Configuration configuration)
-            throws IOException {
-        final String sqlContent = String.join("", statements);
         DefaultContext defaultContext =
                 new DefaultContext(
-                        new Configuration(configuration)
+                        new Configuration(MINI_CLUSTER.getClientConfiguration())
                                 // Make sure we use the new cast behaviour
                                 .set(
                                         ExecutionConfigOptions.TABLE_EXEC_LEGACY_CAST_BEHAVIOUR,
                                         ExecutionConfigOptions.LegacyCastBehaviour.DISABLED),
                         Collections.emptyList());
+        executor =
+                Executor.create(
+                        defaultContext,
+                        InetSocketAddress.createUnresolved(
+                                SQL_GATEWAY_REST_ENDPOINT_EXTENSION.getTargetAddress(),
+                                SQL_GATEWAY_REST_ENDPOINT_EXTENSION.getTargetPort()),
+                        "test-session");
+    }
+
+    @AfterEach
+    public void afterEach() {
+        executor.close();
+    }
 
-        InputStream inputStream = new ByteArrayInputStream(sqlContent.getBytes());
+    @Override
+    protected String runSingleStatement(String statement) throws Exception {
+        String inputStatement = statement + ";\n";
+        InputStream inputStream = new ByteArrayInputStream(inputStatement.getBytes());
         ByteArrayOutputStream outputStream = new ByteArrayOutputStream(256);
 
-        try (final Executor executor =
-                        Executor.create(
-                                defaultContext,
-                                InetSocketAddress.createUnresolved(
-                                        SQL_GATEWAY_REST_ENDPOINT_EXTENSION.getTargetAddress(),
-                                        SQL_GATEWAY_REST_ENDPOINT_EXTENSION.getTargetPort()),
-                                "test-session");
-                Terminal terminal = new DumbTerminal(inputStream, outputStream);
+        try (Terminal terminal = new DumbTerminal(inputStream, outputStream);
```

Comment:
It's a little different from the actual usage. Actually, we bind the Executor to a terminal in the SqlClient. Do you think whether it's better we overwrite the runStatements in the base class?